### PR TITLE
docs: adding setup flow

### PR DESCRIPTION
**Pull request by DanielHe4rt**
## Description

Adds a step-by-step guide to run the Rust-RO server locally. 

- [x] 1. Config.json 
- [x] 2. Setup Database 
- [x] 3. Building Binaries
- [x] 4. Running the Binaries
- [x] Extra: Remodeling some aspects from the README.

Hope you like the modifications. 

Cheers!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,38 +80,221 @@ Checkout [architectures notes](doc/Architecture.md)
...
+```
+
+Simple like that! Now you're good to run your servers. 
+
+### 5.4 Running the Binaries
+
+After we have everyting set-up (binaries and database), we should run these binaries to make the game playable:
+
+- server
+- maps
```
Comment (nmeylan) [inline README.md:214]: this is optional, if some maps are missing from mapcache

File: README.md [hunk 2]
```diff
@@ -41,38 +80,221 @@ Checkout [architectures notes](doc/Architecture.md)
...
+- packets
```
Comment (nmeylan) [inline README.md:215]: running packets tool is not necessary, it is only for when developer add new packet

File: README.md [hunk 3]
```diff
@@ -1,25 +1,64 @@
 ![build](https://github.com/nmeylan/rust-ro/actions/workflows/rust.yml/badge.svg)
-# What is rust-ro?
-rust-ro is a Ragnarok mmo server implementation written in rust. While it is a from scratch implemention it is heavily inspired by [herculesWS](https://github.com/HerculesWS/Hercules) and [rathena](https://github.com/rathena/rathena), for example this implementation support same scripting language for NPC meaning that existing scripts should work on this implementation and use the same database structure than rathena and some naming convention were kept.
 
-## Ambition
+![Rust-RO logo](.github/images/logo.png)
```
Comment (nmeylan) [inline README.md:3]: I'm not sure i want to add a logo right now, if you can remove it
Nice artwork by the way

File: README.md [hunk 4]
```diff
@@ -41,38 +80,221 @@ Checkout [architectures notes](doc/Architecture.md)
...
+## 5. Setup 
 
-# Usage
-*This section will be rewritten*
+Here's a list of **pre-requisites** to run rust-ro:
 
-**Pre-requisite:**
-- postgresql (rathena db was converted using pgloader and then adapted)
-- RO db `db/pg.sql`
+* Docker OR PostgreSQL 16+ directly on your machine
```
Comment (nmeylan) [inline README.md:88]: i think minimal postgresql version could be 12

File: README.md [hunk 5]
```diff
@@ -41,38 +80,221 @@ Checkout [architectures notes](doc/Architecture.md)
...
+-- In case you want to take a look: 
+-- /rust-ro/docker/volumes/init.sql
+
+CREATE USER ragnarok WITH PASSWORD 'ragnarok';
+CREATE DATABASE ragnarok;
+GRANT ALL PRIVILEGES ON DATABASE ragnarok TO ragnarok;
+ALTER DATABASE ragnarok OWNER TO ragnarok;
+```
+
+After your database is up, you will need import the Game SQL (`/rust-ro/pg.sql`) to our new database. For that follow the steps:
```
Comment (nmeylan) [inline README.md:150]: normally all step below should be automated, if it is not the case we can create an issue for this, but leaving your input for now

File: README.md [hunk 6]
```diff
@@ -41,38 +80,221 @@ Checkout [architectures notes](doc/Architecture.md)
...
+To run the `server` binary, you will need a `ENV` variable called `DATABASE_PASSWORD` together with your command:
 
+```shell
+DATABASE_PASSWORD=ragnarok cargo run --bin=server
+```
 
+If everything goes right, you should receive something like this output:
+
+```
+exclude pattern sqlx
```
Comment (nmeylan) [inline README.md:229]: from line 229 to237 could be remove as it is not supposed to log those lines (it was for debugging purpose but it remained)

File: README.md [hunk 7]
```diff
@@ -41,38 +80,221 @@ Checkout [architectures notes](doc/Architecture.md)
...
+INFO [server::server::boot::mob_spawn_loader] load 3391 mob spawns in 0.022 secs
+INFO [server] load 897 map-cache in 0.05 secs
+INFO [server::proxy] Start proxy for Char proxy, 6123:6121
+INFO [server::proxy] Start proxy for map proxy, 6124:5121
+INFO [server::server] Server listen on 0.0.0.0:6901
+```
+
+#### 5.4.2 Running the Game
+
+TODO: roBrowser or what?
```
Comment (nmeylan) [inline README.md:249]: as i support only a specific packetver for the moment i think robrowser is the only viable option (unless someone is able to build a client supporting packetver 20120307

**Comment by nmeylan** [comment]
Wow thanks a lot for your contribution, very helpful, i have made small comment which should takes fews minutes to fix

File: README.md (hunk 5 above)
Comment (DanielHe4rt) [reply README.md:150]: When I tried to install it didn't worked as expected. Maybe missed something? IDK but we can improve this process later :D

File: README.md (hunk 7 above)
Comment (DanielHe4rt) [reply README.md:249]: Hmm I don't think it is a possibility in a short-term period. I'll need some feedbacks to understand how you configured your roBrowser and make a tutorial of it. 

Yesterday I figured out how to run but there's some glitch stuff and Idk if it's related to the .grf files or if I misconfigured something.

File: README.md (hunk 3 above)
Comment (DanielHe4rt) [reply README.md]: Done!

File: README.md (hunk 7 above)
Comment (DanielHe4rt) [reply README.md]: Decided to leave a small text with some print to make people more interested on the game running together with rust-ro.

**Comment by DanielHe4rt** [comment]
@nmeylan All good IHMO.

**Pull request merged**